### PR TITLE
add hasLabel and hasIcon prop to StyledButtonKind

### LINKED ISSUE #5613: iconProp and labelProp extend props in theme for Button
I would like to extend css for specific conditions when the icon and label props has been passed to the Grommet Button, but observed that there are no Grommet specific props available. It would be very helpful to me to have these props.

### Expected Behavior

There is iconProp and labelProp available for extending CSS.

### Actual Behavior

The props are not available.

### URL, screen shot, or Codepen exhibiting the issue

image

### Steps to Reproduce

Console log out the extend props.

- Grommet version: 2.16.2
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): Mac desktop

### PR DESCRIPTION
**Pull request by lleBit**
#### What does this PR do?
add missing hasIcon and hasLabel to StyledButtonKind. Additionally, sets the line-height to 0 when there is only an icon.

#### Where should the reviewer start?
Button.js

#### What testing has been done on this PR?
Manual + added a unit test.

#### How should this be manually tested?
1. Console log out the props under StyledButtonKind.
2. Console log out the button extend props in the jest test.

#### Any background context you want to provide?
I need these props to style the theme of a button that only has an icon

#### What are the relevant issues?
hasIcon and hasLabel is accessible to StyledButton already.
closes #5613 

#### Do the grommet docs need to be updated?
I don't think so

#### Should this PR be mentioned in the release notes?
maybe no

#### Is this change backwards compatible or is it a breaking change?
non-breaking changes

image

### CODE REVIEW

File: src/js/components/Button/__tests__/Button-test.js [hunk 1]
```diff
@@ -569,4 +569,18 @@ describe('Button', () => {
 
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  test(`line-height should be zero for icon only`, () => {
+    const { getByRole } = render(
+      <Grommet
+        theme={{
+          button: { default: {} },
+        }}
```
Comment (jcfilben) [inline src/js/components/Button/__tests__/Button-test.js:578]: Button should not have default defined in the theme. Default should only be defined if using button kind.

```suggestion
```

Comment (lleBit) [reply src/js/components/Button/__tests__/Button-test.js]: @jcfilben - Thanks for catching that! Removed the theme.

**Review by jcfilben — COMMENTED** [review]
Screen Shot 2021-09-27 at 2 03 17 PM
This change doesn't look great. The Buttons with text are not aligned with the Button with only an icon. I'm wondering if we should remove this:

```
${(props) =>
    props.hasIcon &&
    !props.hasLabel &&
    `
    line-height: 0;
  `}
```

Is there a reason we want to set the line-height to 0 when a Button only contains an icon?

**Comment by lleBit** [comment]
> Is there a reason we want to set the line-height to 0 when a Button only contains an icon?

@jcfilben - I copied that code from StyledButton.js. I figured it would make it consistent with the behavior that already exists.

It's not a big deal since I can override it on my side of the code. StyledButton.js and StyledButtonKind.js wouldn't behave the same, but maybe that's what you guys want. Let me know.

**Comment by halocline** [comment]
> > Is there a reason we want to set the line-height to 0 when a Button only contains an icon?
> 
> @jcfilben - I copied that code from StyledButton.js. I figured it would make it consistent with the behavior that already exists.
> 
> It's not a big deal since I can override it on my side of the code. StyledButton.js and StyledButtonKind.js wouldn't behave the same, but maybe that's what you guys want. Let me know.

Thank you for checking for parity between the two. In this case though, I do not think the line-height adjustment is needed (it was added to [fix an alignment issue particular to StyledButton](https://github.com/grommet/grommet/pull/2770), but that issue is not present in StyledButtonKind).

Let's revert that adjustment which should cleanup the snapshot changes. Everything else is looking great.

**Comment by lleBit** [comment]
> Let's revert that adjustment which should cleanup the snapshot changes. Everything else is looking great.
@halocline 

Sounds good. I just removed it and also the test. I'm not sure how to test this since it's a prop that gets passed to StyledButtonKind. Thank you!!

**Comment by halocline** [comment]
> Sounds good. I just removed it and also the test. I'm not sure how to test this since it's a prop that gets passed to StyledButtonKind. Thank you!!

I think in this particular case I would be comfortable without a supporting test, because it is more of internal implementation detail. @jcfilben thoughts?

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**